test/Quake/const_prop_complex.qke

```diff
@@ -0,0 +1,133 @@
+// ========================================================================== //
+// Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                 //
+// All rights reserved.                                                       //
+//                                                                            //
+// This source code and the accompanying materials are made available under   //
+// the terms of the Apache License 2.0 which accompanies this distribution.   //
+// ========================================================================== //
+
+// RUN: cudaq-opt -const-prop-complex %s | FileCheck %s
+
+func.func @__nvqpp__mlirgen__function_test_complex_constant_array._Z27test_complex_constant_arrayv() attributes {"cudaq-entrypoint", "cudaq-kernel", no_this} {
+    %cst = arith.constant 0.000000e+00 : f32
+    %cst_0 = arith.constant 0.70710678118654757 : f64
+    %0 = arith.truncf %cst_0 : f64 to f32
```

Comment:
`cc.cast`?

Reply:
I couldn't generate code from examples that uses a cc.cast here, so no idea how to properly test that. I'd prefer to leave it for another PR that moves all of this to the canonicalizer and replaces arith.truncf with cc.cast.

Reply:
Added an issue.

Reply:
https://github.com/NVIDIA/cuda-quantum/issues/1955

Also added a cast const prop pattern and a quake test